sxr-controller/app/src/main/java/com/samsungxr/sample/controller/SampleMain.java

```diff
@@ -103,6 +104,9 @@ public void onCursorControllerSelected(SXRCursorController newController, SXRCur
                 newController.setCursorDepth(DEPTH);
                 newController.setCursorControl(SXRCursorController.CursorControl.PROJECT_CURSOR_ON_SURFACE);
                 newController.getPicker().setEventOptions(eventOptions);
+                if (controller instanceof SXRGazeCursorController) {
+                    ((SXRGazeCursorController)controller).setTouchScreenDepth(DEPTH);
```

Comment:
Add controller.setCursorControl(SXRCursorController.CursorControl.CURSOR_CONSTANT_DEPTH) and the drag will be smooth since the position will be continuously updated.

There is more to it because the dragged object can be dropped if you move your finger too fast but that is a separate problem.

Reply:
In my opinion, the SXRCursorController/SDK should not stop the drag never.
Just the application side should be allowed to stop the drag.

Reply:
Some times the SDK never stop the drag if you keep doing a lot of "double touch" on the selected object. 
The selected object gets glued to the cursor for ever.

Reply:
The SDK doesn't stop the drag on its own. The app controls the start and the stop. The CURSOR_CONSTANT_DEPTH is needed so the transform gets updated every frame, per implementation. Double-touch is probably again something needs to be addressed in the events listener on the app side. See the mPickHandler in sxr-controller.